doc/main/api/celeritas.rst

```diff
@@ -75,6 +75,51 @@ On-device access
 
 .. doxygenclass:: celeritas::PhysicsTrackView
 
+Propagation and magnetic field
+------------------------------
+
+The propagation interface is built on top of the geometry to allow both curved
+and straight-line movement. Field propagation is based on a composition of:
+
+Field
+  Maps a point in space and time to a field vector.
+Equation of motion
+  Calculates the path derivative of position and momentum given their current
+  state and the templated field.
+Integrator
+  Numerically integrates a new position/momentum state given the start,
+  path derivative, and step length.
+Driver
+  Integrate path segments that satisfy certain error conditions, solving for
+  the required segment length.
+Propagator
+  Given a maximum physics step, advance the geometry state and momentum along
+  the field lines, satisfying given constraints for the maximum geometry error.
```

Comment:
Do we document somewhere what is `certain error conditions` and how to give the 'constraints'?

Reply:
The detailed field propagator documentation gives more details. I can also link to the field propagator options.

Reply:
> I can also link to the field propagator options.

+1